### PR TITLE
Add bigip_device_auth_radius module to enable RADIUS auth

### PR DESCRIPTION
**Pull request by andreykashcheev**
### Summary 

This change introduces new bigip_device_auth_radius module intended for enabling RADIUS auth

### Chnages

 - Add new module titled as bigip_device_auth_radius
 - InProgress -> Add unit tests for new module
 - InProgress -> Add Functional tests for new module

### Additional Information

### CODE REVIEW

File: test/units/modules/network/f5/test_bigip_device_auth_radius.py [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+except ImportError:
+    from ansible_collections.f5networks.f5_modules.plugins.modules.bigip_device_auth_radius import ApiParameters
+    from ansible_collections.f5networks.f5_modules.plugins.modules.bigip_device_auth_radius import ModuleManager
+    from ansible_collections.f5networks.f5_modules.plugins.modules.bigip_device_auth_radius import ArgumentSpec
+
+    # Ansible 2.8 imports
+    from units.compat import unittest
+    from units.compat.mock import Mock
+
+    from units.modules.utils import set_module_args
```
Comment (wojtek0806) [inline test/units/modules/network/f5/test_bigip_device_auth_radius.py]: incorrect file was used, not templated, copy paste is unacceptable review the template and correct these

File: library/modules/bigip_device_auth_radius.py [hunk 2]
```diff
@@ -0,0 +1,754 @@
...
+                    'status': ['stableinterface'],
+                    'supported_by': 'certified'}
+
+DOCUMENTATION = r'''
+---
+module: bigip_device_auth_radius
+short_description: Manage RADIUS authentication on a BIG-IP
+description:
+  - Manage the RADIUS authentication configuration.
+version_added: 2.9
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: use template this is incorrect, review the templated files and correct the version

File: library/modules/bigip_device_auth_radius.py [hunk 3]
```diff
@@ -0,0 +1,754 @@
...
+      secret:
+        description:
+          - Specifies secret used for accessing RADIUS server.
+        type: str
+      timeout:
+        description:
+          - Specifies the timeout value in seconds.
+          - The default timeout value is C(3) seconds.
+        type: int
+        default: 3
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: why is there a default? ansible modules do not set defaults on items sent to the device

File: library/modules/bigip_device_auth_radius.py [hunk 4]
```diff
@@ -0,0 +1,754 @@
...
+          - This field is required.
+            In that case, the simple list can specify server IPs. See examples for
+            more clarification.
+        type: str
+      port:
+        description:
+          - The port of the server.
+          - If no port number is specified, the default port C(1812) is used.
+        type: int
+        default: 1812
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: no defaults

File: library/modules/bigip_device_auth_radius.py [hunk 5]
```diff
@@ -0,0 +1,754 @@
...
+      - Enables or disables validation of the accounting response vector.
+      - This option should be necessary only on older servers
+    type: str
+    default: disabled
+  retries:
+    description:
+      - Specifies the number of authentication retries that the BIG-IP local traffic management system allows before authentication fails.
+      - The default value is C(3).
+    type: int
+    default: 3
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: no defaults

File: library/modules/bigip_device_auth_radius.py [hunk 6]
```diff
@@ -0,0 +1,754 @@
...
+  service_type:
+    description:
+      - Specifies the type of service requested from the RADIUS server. The default value is authenticate-only
+    type: str
+    default: authenticate-only
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: no defaults

File: library/modules/bigip_device_auth_radius.py [hunk 7]
```diff
@@ -0,0 +1,754 @@
...
+            response = resp.json()
+        except ValueError as ex:
+            raise F5ModuleError(str(ex))
+
+        if 'code' in response and response['code'] == 400:
+            if 'message' in response:
+                raise F5ModuleError(response['message'])
+            else:
+                raise F5ModuleError(resp.content)
+
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: incorrect patterns

File: library/modules/bigip_device_auth_radius.py [hunk 8]
```diff
@@ -0,0 +1,754 @@
...
+        resp = self.client.api.get(uri)
+        try:
+            response = resp.json()
+        except ValueError as ex:
+            raise F5ModuleError(str(ex))
+        if 'code' in response and response['code'] == 400:
+            if 'message' in response:
+                raise F5ModuleError(response['message'])
+            else:
+                raise F5ModuleError(resp.content)
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: as as above

File: library/modules/bigip_device_auth_radius.py [hunk 9]
```diff
@@ -0,0 +1,754 @@
...
+        try:
+            response = resp.json()
+        except ValueError as ex:
+            raise F5ModuleError(str(ex))
+
+        if 'code' in response and response['code'] in [400, 403]:
+            if 'message' in response:
+                raise F5ModuleError(response['message'])
+            else:
+                raise F5ModuleError(resp.content)
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: change

File: library/modules/bigip_device_auth_radius.py [hunk 10]
```diff
@@ -0,0 +1,754 @@
...
+
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: change

File: library/modules/bigip_device_auth_radius.py [hunk 11]
```diff
@@ -0,0 +1,754 @@
...
+        try:
+            response = resp.json()
+        except ValueError as ex:
+            raise F5ModuleError(str(ex))
+
+        if 'code' in response and response['code'] == 400:
+            if 'message' in response:
+                raise F5ModuleError(response['message'])
+            else:
+                raise F5ModuleError(resp.content)
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: change

File: library/modules/bigip_device_auth_radius.py [hunk 12]
```diff
@@ -0,0 +1,754 @@
...
+
+        try:
+            response = resp.json()
+        except ValueError as ex:
+            raise F5ModuleError(str(ex))
+        if 'code' in response and response['code'] == 400:
+            if 'message' in response:
+                raise F5ModuleError(response['message'])
+            else:
+                raise F5ModuleError(resp.content)
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: change

File: library/modules/bigip_device_auth_radius.py [hunk 13]
```diff
@@ -0,0 +1,754 @@
...
+            try:
+                response = resp.json()
+                servers.append(resp.json())
+            except ValueError as ex:
+                raise F5ModuleError(str(ex))
+            if 'code' in response and response['code'] == 400:
+                if 'message' in response:
+                    raise F5ModuleError(response['message'])
+                else:
+                    raise F5ModuleError(resp.content)
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: incorrect

File: library/modules/bigip_device_auth_radius.py [hunk 14]
```diff
@@ -0,0 +1,754 @@
...
+            self.client.provider['server_port'],
+        )
+        resp = self.client.api.get(uri)
+        try:
+            response = resp.json()
+        except ValueError:
+            return False
+        if resp.status == 404 or 'code' in response and response['code'] == 404:
+            return False
+        return True
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: incorrect

File: library/modules/bigip_device_auth_radius.py [hunk 15]
```diff
@@ -0,0 +1,754 @@
...
+            servers=dict(
+                type='list',
+                elements='dict',
+                options=dict(
+                    server=dict(
+                        required=True
+                    ),
+                    port=dict(
+                        type='int',
+                        default=1812
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: no defaults

File: library/modules/bigip_device_auth_radius.py [hunk 16]
```diff
@@ -0,0 +1,754 @@
...
+            retries=dict(
+                default=3,
+                type='int'
+            ),
+            service_type=dict(
+                choices=[
+                    'authenticate-only', 'login', 'default', 'framed', 'callback-login', 'callback-framed', 'outbound', 'administrative',
+                    'nas-prompt', 'callback-nas-prompt', 'call-check', 'callback-administrative'
+                ],
+                default='authenticate-only'
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: no defaults

File: library/modules/bigip_device_auth_radius.py [hunk 17]
```diff
@@ -0,0 +1,754 @@
...
+        result['auth_source'] = self.read_current_auth_source_from_device()
+        return self.get_api_parameters(params=result)
+
+
+class ArgumentSpec(object):
+    def __init__(self):
+        self.supports_check_mode = True
+        argument_spec = dict(
+            retries=dict(
+                default=3,
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: no defaults

File: library/modules/bigip_device_auth_radius.py [hunk 18]
```diff
@@ -0,0 +1,754 @@
...
+        return UsableChanges(params=params)
+
+    def get_reportable_changes(self, params=None):
+        return ReportableChanges(params=params)
+
+    def get_module_parameters(self, params=None):
+        return ModuleParameters(params=params)
+
+    def get_api_parameters(self, params=None):
+        return ApiParameters(params=params)
```
Comment (wojtek0806) [inline library/modules/bigip_device_auth_radius.py]: why is this here?

**Comment by andreykashcheev** [comment]
I am going to close this PR and will re-open a new one with correct changes
